### ISSUE TITLE
open_graph.py's _create_absolute_fileurl  method does not honor a SITEURL with a subdirectory

### ISSUE THREAD

**Issue by fluid217** [pelican-plugins/seo#66]
Given 

`SITEURL = 'mysite.com/blog'` in the Pelican settings file.
and a `fileurl` of 'posts/my_article/index.html`

the `_create_absolute_fileurl`  method in `open_graph.py` will return a `file_url` missing the `SITEURL`s subdirectory.
This is due to the way that urllib's `parse.urljoin` works. 

My quick workaround for now is to append a slash to the `siteurl`. If it's redundant, it will be stripped out by `urljoin`

```
def __init__(
       self, siteurl, fileurl, file_type, title, description, image, locale
      ) -> None:
          self.siteurl = siteurl + "/"
```

**Comment by buanzo** [comment]
I am having the same issue. My Pelican is in a subfolder of my domain, example.com/blog, and the image/og_image attributes begin with /images/... - The resulting image url lacks the /blog before /images/.

**Comment by buanzo** [comment]
Maybe something like this, to avoid making sure SITEURL ends with '/':

```python
def _create_absolute_fileurl(self):
    """Join site URL and file path."""
    # Ensure fileurl is a relative path
    if self.fileurl.startswith('/'):
        relative_fileurl = self.fileurl[1:]
    else:
        relative_fileurl = self.fileurl

    file_url = parse.urljoin(self.siteurl, relative_fileurl)
    return file_url
```

**Comment by MaevaBrunelles** [comment]
Hello @buanzo, 

Feel free to open a PR with the patch and tests that ensure the canonical URL is well generated :)

**Comment by buanzo** [comment]
Absolutely! I'm working on the injector plugin, then tackle some bugs. Is there a devs mailing list ?

**Comment by max-pfeiffer** [comment]
+1 here. I just saw in my articles HTML source that I ran into this bug as well.

@justinmayer If this repo here is still maintained I would quickly. provide a fix for this.

**Comment by justinmayer** [comment]
Hi Max. I think you can see that the last release was published to PyPI just two months ago, so perhaps it is safe to assume that this repository is still maintained? 😉 

As Maëva already said above, pull requests that address the issue and that include corresponding tests would be most welcome 😊

**Comment by max-pfeiffer** [comment]
@justinmayer here you go 😄  : https://github.com/pelican-plugins/seo/pull/74
